localization/pose_initializer/src/pose_initializer/pose_initializer_core.cpp

```diff
@@ -57,6 +57,31 @@ PoseInitializer::PoseInitializer() : Node("pose_initializer")
   logger_configure_ = std::make_unique<tier4_autoware_utils::LoggerLevelConfigure>(this);
 
   change_state(State::Message::UNINITIALIZED);
+
+  if (declare_parameter<bool>("user_defined_initial_pose.enable")) {
+    const auto initial_pose_array =
+      declare_parameter<std::vector<double>>("user_defined_initial_pose.pose");
+    if (initial_pose_array.size() != 7) {
+      throw std::invalid_argument(
+        "Could not set user defined initial pose. The size of initial_pose is " +
+        std::to_string(initial_pose_array.size()) + ". It must be 7.");
+    } else if (
+      initial_pose_array[3] < 1e-6 && initial_pose_array[4] < 1e-6 &&
+      initial_pose_array[5] < 1e-6 && initial_pose_array[6] < 1e-6) {
+      throw std::invalid_argument("Input quaternion is invalid. All elements are close to zero.");
```

Comment:
Strictly speaking, is it possible for a quaternion to have w about 0, others negative, and norm about 1?

Reply:
Thank you for pointing it out.
I modified it to check the absolute value because quaternions can take negative values. https://github.com/autowarefoundation/autoware.universe/pull/6692/commits/ac8185a9c397a66aed1140ca725e4a0e53a1cbea